pkg/cmd/target/options.go

```diff
@@ -0,0 +1,89 @@
+/*
+SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package target
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+)
+
+// TargetKind is representing the type of things that can be targeted
+// by this cobra command. While this may sound stuttery, the alternative
+// of just calling it "Kind" is even worse, hence the nolint.
+// nolint
+type TargetKind string
+
+const (
+	TargetKindGarden  TargetKind = "garden"
+	TargetKindProject TargetKind = "project"
+	TargetKindSeed    TargetKind = "seed"
+	TargetKindShoot   TargetKind = "shoot"
+)
+
+var (
+	AllTargetKinds = []TargetKind{TargetKindGarden, TargetKindProject, TargetKindSeed, TargetKindShoot}
+)
+
+// Options is a struct to support target command
+type Options struct {
+	base.Options
+
+	// Kind is the target kind, for example "garden" or "seed"
+	Kind TargetKind
+	// TargetName is the object name of the targeted kind
+	TargetName string
+}
+
+// NewOptions returns initialized Options
+func NewOptions(ioStreams genericclioptions.IOStreams) *Options {
+	return &Options{
+		Options: base.Options{
+			IOStreams: ioStreams,
+		},
+	}
+}
+
+// Complete adapts from the command line args to the data required.
+func (o *Options) Complete(f util.Factory, cmd *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return errors.New("expected exactly 2 arguments")
+	}
+
+	o.Kind = TargetKind(strings.TrimSpace(args[0]))
+	o.TargetName = strings.TrimSpace(args[1])
+
+	return nil
+}
```

Comment:
have you thought about using hierarchical target commands instead?
something like this structure
```yaml
cmds:
- target:
    # flags: # need to think about that
    # - garden
    # - seed
    # - project
    # - shoot
    cmds:
    - garden:
        flags: ~
    - seed:
        flags:
        - garden
    - project:
        flags: 
        - garden
    - shoot:
        flags:
        - garden
        - seed
        - project
- ssh: 
# [...]
```

Reply:
Seeing how simple the command logic is now, I don't think it would offer many benefits when we split it into subcommands. I was taking a quick glance at cobra's auto completion magic and it seemed like we should be able to offer nice completion even if it's a single command. But that was just a very quick glance ^^

Reply:
Still, I think I would prefer to have it split up into subcommands. You get rid of the switch, don't need to mess around with custom completion logic and parts of the options validation is also obsolet as it is handled by cobra. Let's also talk how it would look like if we evaluate also the flags, e.g. `g target shoot -g canary -p core funny` once you are back

Reply:
Let's split this after this PR has been merged. The split should not affect the general structure of the `target` package and with this PR merged, we have a basis for other commands to be implemented.